src/lib.rs

```diff
@@ -259,14 +280,15 @@ impl ModuleConfig {
                 log::trace!("Module::handler_for_host_path: host matched, examining path");
                 // The important detail here is that strip_suffix returns None if the suffix
                 // does not exist. So ONLY paths that end with /... are substring-matched.
-                if r.path
+                let route_match = r
```

Comment:
Say we have three modules defined:

```toml
[[module]]
route = "/..."

[[module]]
route = "/foo/..."

[[module]]
route = "/foo/bar/..."
```

Now say we get a request for `/foo/bar/baz`.

Which route will match?

Reply:
I checked and even BTreeSet won't do this for us, so I am going to use the https://github.com/bluss/indexmap crate.